### PR TITLE
Better tree shaking with specific imports from lodash

### PR DESCRIPTION
**Pull request by BahjatSalem**
Instead of doing (for example)
`import * as _ from 'lodash';`
we're doing a 
`import defaultsDeep from 'lodash-es/defaultsDeep';`
and then using `defaultsDeep` directly. This removes unnecessary code imports from `lodash`.

In addition, we're using the `importHelpers` Typescript compiler flag, for helpers like `__extends` to be imported from `tslib` instead of being added to every single JS file where it's needed.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+import alias from 'rollup-plugin-alias';
 
 export default {
   format: 'umd',
   moduleName: 'angular-tree-component',
   plugins: [
     nodeResolve({ jsnext: true, main: true, module: true }),
-    commonjs(),
-    uglify()
+        alias({
```
Comment (adamkleingit) [inline rollup.config.js]: Why is this needed?

**Comment by adamkleingit** [comment]
@BahjatSalem 
Did you check that tree shaking actually works better with this?
When I tried this before the package size was the same and tree shaking didn't remove unused lodash code.

File: rollup.config.js (hunk 1 above)
Comment (BahjatSalem) [inline rollup.config.js]: My bad. I was goret-pushing my commits. It's not needed.

**Comment by BahjatSalem** [comment]
Well it does make a difference in my other project.
If my memories are correct, I managed to reduce the size of `lodash-es` from more than 200 kB to ~50 kB, according to `source-map-explorer`.
I haven't checked the effects on the `angular-tree-component.umd.js` bundle though.

**Comment by ChristianStornowski** [comment]
This commit would be awsm. We are using lodash-es in our project and now we have the big lodash library in our project because we want to use this very good project.

https://github.com/th0r/webpack-bundle-analyzer cloud help to analyze dependencies.
